python_ta/typecheck/base.py

```diff
@@ -219,6 +221,14 @@ def can_unify(self, t1, t2):
         """Return true iff given argument types can be unified."""
         if isinstance(t1, TypeVar) or isinstance(t2, TypeVar):
             return True
+        elif isinstance(t1, GenericMeta) and isinstance(t2, GenericMeta):
+            if not _geqv(t1, t2):
+                return False
+            elif t1.__args__ is not None and t2.__args__ is not None:
+                for a1, a2 in zip(t1.__args__, t2.__args__):
+                    if not self.can_unify(a1, a2):
+                        return False
+                return True
```

Comment:
You should have a final `else` block here; otherwise this method might return `None`.